### DISCUSSION TITLE
Certificate chain in client certificate

### DISCUSSION THREAD

**Discussion by krishnamurthy-dv** [strongswan/strongswan#1347] [Q&A]
Hi 
 We are using strongswan IKE version 5.7.1. In one of our configurations, we have client certificate having a chain of certificates.  We are specifying the client.crt path in the onnections.<conn>.local.cert.file option. But the entire chain is not getting passed to the responder.  Is there any configuration option to enable initiator to load and send the entire certificate chain ?

**Comment by strongX509** [comment]
With `local.certs` or `local.cert.file` only end entity certificates are configured. The CA certificates forming the trust chain are either stored in `swanctl/x509ca` or the path is defined in the `authorities` section of `swanctl.conf`.

**Reply by krishnamurthy-dv** [reply]
We have a client certificate looking like this "end entity cert-> site CA -> tenant CA --> root CA".  Can IKE initiator read this client certificate and send to the responder IKE for authentication?

**Reply by tobiasbrunner** [reply]
See [this FAQ entry](https://docs.strongswan.org/docs/5.9/support/faq.html#_x_509_certificate_chain_files), i.e. you have to provide the intermediate CA certificates in separate files.

**Reply by krishnamurthy-dv** [reply]
Thanks @tobiasbrunner. Adding the CA chain to cacerts path helped.
